### PR TITLE
Update utils.py: fix nan

### PR DESCRIPTION
**Pull request by MengAiDev**
# What does this PR do?
- Update utils.py: fix nan
- Suggested in https://github.com/huggingface/transformers/pull/40093/commits/7e230e5464705e6e235aec47d8a7312a3404ba15

 @ArthurZucker @zucchini-nlp

### CODE REVIEW

**Review by zucchini-nlp — COMMENTED** [review]
Not sure if we want to fix it here. Unless there is an overflow in model code. there should be non NaNs. @gante WDYT

**Comment by gante** [comment]
Hi @MengAiDev 👋 Thank you for opening this issue 🤗 

Agreed with @zucchini-nlp. Before considering code changes, we need to understand the problem -- even if we end up merging your suggestion as-is :)

Questions:
1. Can you share a short stand-alone reproducer for the issue? If not (e.g. private model), can you share which operation introduces the `nan`?
2. Have you tried running generate with `remove_invalid_values=True`? (this removes `nan` in the logits)

**Comment by MengAiDev** [comment]
I see, but this is just I try to fix the circle ci's failed, and I actually don't have the way to propose it., Sorry.
